src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -734,4 +738,15 @@ CHIP_ERROR OTARequestor::SendNotifyUpdateAppliedRequest(OperationalDeviceProxy &
     return cluster.InvokeCommand(args, this, OnNotifyUpdateAppliedResponse, OnNotifyUpdateAppliedFailure);
 }
 
+// Invoked when the device becomes commissioned
+void OTARequestor::OnCommissioningCompleteRequestor(const DeviceLayer::ChipDeviceEvent * event, intptr_t arg)
+{
+    ChipLogProgress(SoftwareUpdate, "Device commissioned, query the default provider");
+
+    // TODO: Should we also send UpdateApplied here?
+
+    // Query the default provider. At the end of this query/update process the Default Provider timer is started
+    (reinterpret_cast<OTARequestor *>(arg))->mOtaRequestorDriver->DriverSendQuery();
```

Comment:
If this is just after initial commissioning, not delayed here may impact further administrative/initial configuration an admin may be trying to do after CommissioningComplete (since commissioning complete doesn't necessarily mean "done configuring", just "done some critical configs to be part of a fabric"). Suggest deferring this.

Reply:
Maybe a new API in the driver class `NotifyOnCommissioned` which internally sets a timer before querying? Or directly calling ScheduleDelayedAction() here? Although I'm not sure if that latter API should be called outside of the driver.

Reply:
What would be a reasonable delay, 30 seconds?

Reply:
Done